sunspot/lib/sunspot/dsl/scope.rb

```diff
@@ -14,7 +14,12 @@ def initialize(scope, setup) #:nodoc:
 
       # Build a restriction to return only fields of the type in the results.
       def field_list(*args)
-        list = args.flatten.map { |field| @setup.field(field.to_sym).indexed_name.to_sym }
+        # if field list is * then all fields
+        if(args.length == 1 && args[0] == '*')
+          list = ['*']
```

Comment:
does this really relate to the PR subject?

Reply:
@serggl  Sorry, It doesn't. Please let me fix it.

Reply:
`if args == ['*']` should do the same :)